### PR TITLE
Make RobotAgent updates run in background

### PR DESCRIPTION
**Pull request by hello-atharva**
## Description

This PR adds a feature where Stretch runs `RobotAgent` updates in the background as it moves around, significantly speeding up long-horizon tasks. It uses modified `slam_toolbox` package to compute a pose graph. Each vertex in the graph is matched with an RGBD reading to retroactively build a consistent voxel map.

This PR is still a work in progress and needs fixes related to threaded variable access.

To use the modified `slam_toolbox` code, execute on your robot:

```
cd ~/ament_ws/src
git clone git@github.com:hello-atharva/slam_toolbox.git -b humble
cd ..
colcon build
source install/setup.bash
```

Then proceed to launch the `stretch_ros2_bridge` server as usual.

## Checklist

- \[ \] I have performed a self-review of my code
- \[ \] If it is a core feature, I have added thorough tests
- \[ \] I have added documentation for the changes
- \[ \] I have updated the README file if necessary
- \[ \] I have run on hardware if necessary

## Screenshots (if applicable)

Add any relevant screenshots or screen recordings to help reviewers understand the changes.

## Additional context

Add any other context or information about the pull request here.

### CODE REVIEW

**Comment by hello-atharva** [comment]
CI should expectedly fail

**Comment by hello-cpaxton** [comment]
- make sure it's a flag so we dont always do it this way (as a flag)
- add installtion instructions to docs

File: src/stretch/agent/robot_agent.py [hunk 1]
```diff
@@ -139,8 +140,21 @@ def __init__(
         if parameters["motion_planner"]["simplify_plans"]:
             self.planner = SimplifyXYT(self.planner, min_step=0.05, max_step=1.0, num_steps=8)
 
+        # Map updates
+        self._robot_lock = Lock()
+        self._update_map_thread = Thread(target=self.update_map_loop)
+        self._update_map_thread.start()
+
+        # Get observations thread
+        self._robot_lock = Lock()
```
Comment (hello-cpaxton) [inline src/stretch/agent/robot_agent.py:149]: we should have a flag, and default this to false, at least initially

we can switch it to default true after more testing + making sure the apps all run properly

File: src/stretch/agent/zmq_client.py [hunk 2]
```diff
@@ -784,6 +789,11 @@ def _update_obs(self, obs):
             if self._iter <= 0:
                 self._iter = max(self._last_step, self._iter)
 
+    def _update_pose_graph(self, obs):
```
Comment (hello-cpaxton) [inline src/stretch/agent/zmq_client.py:792]: type hint

**Review by hello-cpaxton — APPROVED** [review]
LGTM

**Pull request merged**